presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/HivePageSinkMetadataProvider.java

```diff
@@ -54,7 +54,7 @@ public HivePageSinkMetadataProvider(HivePageSinkMetadata pageSinkMetadata, HiveM
         }
         Optional<Partition> modifiedPartition = modifiedPartitions.get(partitionValues);
         if (modifiedPartition == null) {
-            return delegate.getPartition(identity, schemaTableName.getSchemaName(), schemaTableName.getTableName(), partitionValues);
+            return delegate.getPartition(identity, table.get(), partitionValues);
```

Comment:
can table be `empty`?

Reply:
nope., there is if before: `!table.isPresent()`